drop-boxes/register-omero-metadata/register-omero.py

```diff
@@ -159,68 +236,121 @@ def process(transaction):
-	
+
+	print "metadataFile:"
+	print metadataFile
+
+	property_names = getPropertyNames(metadataFile)
+
+	print "property names:"
+	print property_names
+
+	valid_names = validatePropertyNames(property_names)
+	if not valid_names:
+		raise ValueError("Invalid Property Names.")
+
+	#keep track of number of images for openBIS ID
+	image_number = 0
+	#Initialize openBIS imaging experiment
+	imagingExperiment = None
+	previousProps = {}
+	existing_experiment_ids = []
+
+	print "start reading metadata file"
 	# Iterate over the metadata entries containing all pre-specified imaging metadata
 	for line in metadataFile[1:]:  # (Exclude header)
 		# Get modality and other metadata from tsv here for one sample
 		properties = {}
 		
-		# Retrieve the image file name
+		# Retrieve the image file name, please no whitespace characters in filename!
 		fileName = getFileFromLine(line)
-
-		imageFile = os.path.join(incomingPath, fileName)
+		# Due to the datahandler we need to add another subfolder of the same name to the path
+		imageFolder = os.path.join(incomingPath, folderName)
+		imageFile = os.path.join(imageFolder, fileName)
 		print "New incoming image file for OMERO registration:\t" + imageFile
 
 		# 4. After we have received the omero dataset id, we know where to attach the image to
 		# in OMERO. We pass the omero dataset id and trigger the image registration process in OMERO.
 		omero_image_ids = registrationProcess.registerImageFileInOmero(imageFile, omero_dataset_id)
 		print "Created OMERO image identifiers:\t" + str(omero_image_ids)
 
+		omero_failed = len(omero_image_ids) < 1
+		if omero_failed:
+			raise ValueError("Omero did not return expected image ids.")
+
 		# 5. Additional metadata is provided in an own metadata TSV file. 
 		# We extract the metadata from this file.
 		#registrationProcess.extractMetadataFromTSV()
 
+		properties = getPropertyMap(line, property_names)
+		print "Metadata properties:\t"
+		printPropertyMap(properties)
+		
+		#one file can have many images, iterate over all img ids
+		for img_id in omero_image_ids:
+			registrationProcess.registerKeyValuePairs(img_id, filterOmeroPropertyMap(properties))
+
+
+		####
 		# 6. In addition to the image registration and technical metadata storage, we want to add
 		# further experimental metadata in openBIS. This metadata contains information about the 
 		# imaging experiment itself, such as modality, imaged tissue and more. 
 		# We also want to connect this data with the previously created, corresponding OMERO image id t
 		# hat represents the result of this experiment in OMERO. 
-		#registrationProcess.registerExperimentDataInOpenBIS(omero_image_ids)
-
-		# 7. Last but not least we create the open science file format for images which is
-		# OMERO-Tiff and store it in OMERO next to the proprierary vendor format.
-		#registrationProcess.triggerOMETiffConversion()
+		#registrationProcess.registerExperimentDataInOpenBIS(omero_image_ids) # I did it myyy wayyyy
```

Comment:
```suggestion
```

Reply:
nice piece of nostalgia, but I could reluctantly part with it, if andreas agrees